eos_validator.py

```diff
@@ -0,0 +1,51 @@
+
+# Check the TOKEN amount
+def check_balance(node_host="http://localhost:8888", snapshot_csv="./snapshot.csv", eos_total=1000000000):
+    with open(snapshot_csv, 'r') as fp:
+        account_onchain_total = 0
+        for onepersion in fp.readline()
+            account_name, account_amount, pub_key = json.loads(onepersion)['name'], json.loads(onepersion)['amount'],json.loads(onepersion)['pub'] 
+            if account_amount<1.0:
+                continue
+            account_balance = exec('cleos -u ' + node_host + ' get currency balance eosio.token ' + account_name)
+            account_net_delegated = exec('cleos -u ' + node_host + ' get account ' + account_name)['delegated_bandwidth']['net_weight']
+            account_cpu_delegated = exec('cleos -u ' + node_host + ' get account ' + account_name)['delegated_bandwidth']['cpu_weight']
+            account_onchain_amount = account_balance+account_net_delegated+account_net_delegated
+            if abs(account_amount - account_onchain_amount) > 0.0001:
+                # Validate Failed because the amount in onchainblock NOT same with the snapshot
+                raise Exception("account balance failed: ", account_name)
+
+            account_onchain_pubkey = exec('cleos -u ' + node_host + ' get account ' + account_name)['permissions']['owner']
```

Comment:
again? or am i missing somethign?

Reply:
compare the public key from snapshot with the acount onchain's public key to make sure transfer into correct erc20 mapping address